### PR TITLE
Add cni version to userAgent

### PR DESCRIPTION
**Pull request by jchen6585**
**What type of PR is this?**

enhancement

**Which issue does this PR fix**:
#2518 

**What does this PR do / Why do we need it**:
Includes current CNI version in userAgent.

**If an issue # is not available please add repro steps and logs from IPAMD/CNI showing the issue**:

**Testing done on this change**:

Manually tested through looking at cloudwatch logs.

**Automation added to e2e**:

No

**Will this PR introduce any new dependencies?**:

No

**Will this break upgrades or downgrades. Has updating a running cluster been tested?**:

**Does this change require updates to the CNI daemonset config files to work?**:

No

**Does this PR introduce any user-facing change?**:

No

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: charts/aws-vpc-cni/values.yaml [hunk 1]
```diff
@@ -78,6 +78,7 @@ env:
   DISABLE_NETWORK_RESOURCE_PROVISIONING: "false"
   ENABLE_IPv4: "true"
   ENABLE_IPv6: "false"
+  VPC_CNI_VERSION: "v1.15.0"
```
Comment (jchen6585) [inline charts/aws-vpc-cni/values.yaml:81]: Based on what I saw from the implementation of other variables, it seems the other env variable values are eventually hardcoded within the code. This means when upgrading versions we'll have to manually update the default value when calling `GetEnv()` in `session.go`, which defeats the purpose of having this environment variable in the first place.

Comment (jdn5126) [reply charts/aws-vpc-cni/values.yaml:81]: If the environment variable is not defined, we can stick with the behavior as it is today, i.e. not including a version in the user agent string

File: pkg/awsutils/awssession/session.go [hunk 2]
```diff
@@ -89,6 +91,7 @@ func New() *session.Session {
 
 // injectUserAgent will inject app specific user-agent into awsSDK
 func injectUserAgent(handlers *request.Handlers) {
+	version = utils.GetEnv(envVpcCniVersion, "v1.15.0")
```
Comment (jdn5126) [inline pkg/awsutils/awssession/session.go:94]: Let's remove `version` as a global variable and make the default be the empty string, which matches the current behavior.

**Review by jdn5126 — COMMENTED** [review]
Let's also modify https://github.com/aws/amazon-vpc-cni-k8s/blob/master/scripts/generate-cni-yaml.sh#L67 to pass `$VPC_CNI_VERSION`, i.e. `--set env.VPC_CNI_VERSION=$VPC_CNI_VERSION`

**Comment by jdn5126** [comment]
Manual integration test run: https://github.com/aws/amazon-vpc-cni-k8s/actions/runs/6191596936

**Review by jdn5126 — APPROVED** [review]

**Pull request merged**